### PR TITLE
More volume projection fixes

### PR DESCRIPTION
**Pull request by anilyil**
## Purpose

The volume projection routine changes the inverse evaluation into an optimization problem and utilizes the second derivative information to find the u,v,w parameters that correspond to given x,y,z coordinates. However, this second order Newton iteration does not always work; it looks like the final Jacobian ends up producing update vectors that make no sense. I think this might be caused by the second order derivative information.

So in this PR, I removed the contributions of the second order derivatives from the Newton solver, at least temporarily. With this change, the solver works much more robustly for me and @sseraj.

There can be 2 ways I think the second order derivatives are messing up the update: 

1. The resulting optimization problem is non-convex, and therefore exactly evaluating the second order derivatives without any positive definiteness check causes the optimization step to fail.
2. There is a bug in the second derivative code. I documented a similar problem in the curve derivatives in #52 

Ideally, if the reason is the first one listed above, we either need to improve additional checks in the optimization algorithm, or convert the entire thing back to a regular Newton solver. Currently, it looks like it's solving an optimization problem but only using part of the derivative information. Effectively, this is the Newton's method with a weird nonlinear preconditioning step. It does work, but it is not as easy to understand as applying the Newton's method directly.

## Expected time until merged
The fix is fairly quick. I created an issue to track this. I will keep looking into the problem, until then, merging this is fine for me. 

I will dig a bit deeper into this and try to figure out the issue, and then we can test these terms again.

## Type of change

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (non-backwards-compatible fix or feature)
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Documentation update
- [ ] Maintenance update
- [ ] Other (please describe)

## Checklist

- [ ] I have run `flake8` and `black` to make sure the Python code adheres to PEP-8 and is consistently formatted
- [ ] I have formatted the Fortran code with `fprettify` or C/C++ code with `clang-format` as applicable
- [ ] I have run unit and regression tests which pass locally with my changes
- [ ] I have added new tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation

### CODE REVIEW

**Review by sseraj — APPROVED** [review]
The changes are good with me. Once the ADflow tests are retrained, we can merge this and the PR on ADflow at the same time.

**Comment by marcomangano** [comment]
This also looks good to me, but we need one of the maintainers to approve for this to be merged anyway

**Pull request merged**
